### PR TITLE
Spec Review - DO NOT MERGE

### PR DESCRIPTION
**Pull request by davidschramm**
DO NOT MERGE: Review for Rusty only

### CODE REVIEW

File: R/bar-graph-all-years.R [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+    # Define the year as only the year-number in the database name string
+    if (nchar(cchs_str)>= 15)
+    {
+      yr_str <- substring(cchs_str, 5, 13)
+      yr_str <- gsub("_","-", yr_str)
+    } else {
+      yr_str <- substring(cchs_str, 5, 8)
+    }
+    if (k == 1) {
+      cchs_data <- as_label(rwt_data)
```
Comment (DarkShadeKnigh) [inline R/bar-graph-all-years.R:33]: Why as_label?

File: R/bar-graph-all-years.R [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+        # Run one-way ANOVA to determine significance for continuous variables
+        res.aov <- aov(eval(parse(text=paste(var_name_str))) ~ year, data = cchs_data)
+        p <- summary(res.aov)[[1]][["Pr(>F)"]]
+      } else {
+        warning("Variable type is neither categorical nor continuous.")
+      }
+    }
+    # Save significant and ns plots in different folders
+    if (p < 0.05) {
+      img_label <- paste("man/figures/sig_plots/", var_name_str, ".png",
```
Comment (DarkShadeKnigh) [inline R/bar-graph-all-years.R:89]: The location to save them should also be an argument for the overarching function

File: R/bar-graph-all-years.R [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+  } else {
+    summary[i,1] <- var_name_str
+    if (length(p) > 1) {
+      p <- p[1]
+    }
+    summary[i,2] <- p
+  }
+  i <- i+1
+}
+write.csv(summary,"inst/extdata/summary_variable_p-values.csv")
```
Comment (DarkShadeKnigh) [inline R/bar-graph-all-years.R:116]: Passed as argument as well

File: R/string-similarity.R [hunk 4]
```diff
@@ -0,0 +1,56 @@
+library(stringdist)
+library(dplyr)
+setwd("~/GitHub/cchsflow")
+test_different <- read.csv("tests/testdata/test_different.csv", header = FALSE, fileEncoding="UTF-8-BOM")
+test_similar <- read.csv("tests/testdata/test_similar.csv", header = FALSE, fileEncoding="UTF-8-BOM")
```
Comment (DarkShadeKnigh) [inline R/string-similarity.R:5]: Should be function arguments

**Review by rvyuha — CHANGES_REQUESTED** [review]
You should avoid running any code in your R files. Also avoid writing hard paths for outputs if at all possible this limits the usability of your functions.

**Pull request closed without merging**